input: don't cycle pause property, mac: report playback state

Akemi opened the pull request:
this changes some input commands and might break some other use cases. imo PLAY and PAUSE should explicitly set the proper pause state and only PLAYPAUSE should cycle the pause state.

i am not sure if there is a reason for this. the MP_KEY_* events are used in following files, so we need a bit of input of the related persons, @wm4 @philipl @sfan5 @rossy @Dudemanguy 

sry if i forgot someone.

osdep/w32_keyboard.c
video/out/vo_caca.c
video/out/wayland_common.c
video/out/vo_sdl.c
video/out/x11_common.c

wm4 commented:
Not sure if that'd be expected. For example, I used to have a keyboard with a button labeled play/pause, but it certainly wasn't mapped to the X11 equivalent of PLAYPAUSE.

Akemi commented:
that seems kinda bad. at least for consistency. if we don't want to change those defaults i can still set the pause property via the mpv handle/libmpv. though i still find it weird for an explicit PAUSE to cycle the state.

sfan5 commented:
In mpv-android I map `KEYCODE_MEDIA_{PLAY,PAUSE,PLAY_PAUSE}` to the mpv keycodes. Android's [documentation](https://developer.android.com/reference/android/view/KeyEvent.html#KEYCODE_MEDIA_PLAY) just says "Key code constant: Play media key".
I don't know where one would find these keys (TV remotes?) but the new behavior would fit them.

However I can see the old behavior being useful for convenience and "keys aren't the same on platforms"-reasons.

rossy commented:
That makes sense to me from Windows' perspective. I'm surprised the bindings weren't like that already. Microsoft seems pretty explicit about the behaviour of these buttons.

> **APPCOMMAND_MEDIA_PAUSE**
> Pause. If already paused, take no further action. This is a direct PAUSE command that has no state.

> **APPCOMMAND_MEDIA_PLAY**
> Begin playing at the current position. If already paused, it will resume. This is a direct PLAY command that has no state.

philipl commented:
I also don't have a keyboard with these buttons on it. I do have a logitech harmony, which will map keys to whatever you want, but it has separate play and pause buttons. So it's hard to say what would happen in the real world on linux, but there are distinct PLAY, PAUSE, and PLAYPAUSE event codes in the kernel, so it's not crazy to just take them at face value. Presumably people with affected hardware will yell if we make this change.

So perhaps do it and see what breakage gets reported and work from there.

Akemi commented:
i added two now key codes PLAYONLY and PAUSEONLY that do not cycle but explicitly set the pause state. added since they seem to be at least useful for Android, Windows and macOS and it's probably the best course of action (no breakages).

wm4 commented:
Seems like a good solution. Other platforms which want it could make use of it too.

Dudemanguy commented:
Oops, I never commented here. Wayland just uses xkb so whatever breaks on Xorg will break on wayland too. Anyways, I like the newly added keycodes as well.
